### PR TITLE
macos: Initial fixes to get macos building with Makefile

### PR DESCRIPTION
**Pull request by sonneveld**
Also add 'debug' target for.. debugging.
(note: colours are off because allegro patch assumed a different pixel format, to be fixed in a future patch) (or the SDL port)

### CODE REVIEW

File: Engine/Makefile [hunk 1]
```diff
@@ -41,13 +49,18 @@ common.a: $(OBJS_COMMON)
 	@echo $@
 	$(CMD_PREFIX) $(CC) $(CFLAGS) -MD -c -o $@ $<
 
+%.o: %.mm
```
Comment (rofl0r) [inline Engine/Makefile]: do we really want to create dependency information? in my experience this often makes the build almost 2x slower. and the advantage added by this is questionable.

File: Engine/Makefile-objs [hunk 2]
```diff
@@ -70,3 +70,43 @@ APEG = libsrc/apeg-1.2.1/adisplay.c libsrc/apeg-1.2.1/getbits.c libsrc/apeg-1.2.
 AASTR = ../Common/libsrc/aastr-0.1.1/aarot.c ../Common/libsrc/aastr-0.1.1/aastr.c ../Common/libsrc/aastr-0.1.1/aautil.c
 
 AL_MIDI_PATCH = libsrc/allegro-4.2.2-agspatch/midi.c
+
+ALDUMB = libsrc/dumb-0.9.2/allegro/alplay.c \
```
Comment (rofl0r) [inline Engine/Makefile-objs]: why is this needed now ?

Comment (rofl0r) [reply Engine/Makefile-objs]: i just checked my build recipe for ags and it was previously possible to use a system-provided libdumb installation. adding this here would force the usage of the sources shipped with ags, afaict.

File: Engine/Makefile (hunk 1 above)
Comment (sonneveld) [reply Engine/Makefile]: First off, this is to support objective-c code.  We already build dependency information and while it's not perfect, it does help when I'm doing development and incremental builds.

File: Engine/Makefile-objs (hunk 2 above)
Comment (sonneveld) [reply Engine/Makefile-objs]: Ahah, my mistake, will remove.

Comment (sonneveld) [reply Engine/Makefile-objs]: I made it optional (on by default in macos). Since the Windows build uses the builtin source, I want to do the same to pick up on any issues.

File: OSX/patches/allegro-4.4.2.patch [hunk 3]
```diff
@@ -1,5 +1,5 @@
 diff --git a/CMakeLists.txt b/CMakeLists.txt
-index 245be10..105c923 100644
+index 245be10fb..6147a68b3 100644
```
Comment (rofl0r) [inline OSX/patches/allegro-4.4.2.patch]: is this patch still required now that allegro 4.4.3 is released ?

File: Engine/Makefile-defs.linux [hunk 4]
```diff
@@ -1,12 +1,18 @@
+USE_BUILT_IN_LIBSRC = 0
+USE_MIDI_PATCH = 1
```
Comment (rofl0r) [inline Engine/Makefile-defs.linux]: it seems the midi patch is now integrated into allegro as of 4.4.3

File: OSX/patches/allegro-4.4.2.patch (hunk 3 above)
Comment (sonneveld) [reply OSX/patches/allegro-4.4.2.patch]: For macos it is, the patch includes mac specific changes (and 4.4.3)

File: Engine/Makefile-defs.linux (hunk 4 above)
Comment (sonneveld) [reply Engine/Makefile-defs.linux]: This behaviour is unchanged from before. I guess it depends on what version of allegro is bundled with your Linux distro
